neo4j/v1/session.py

```diff
@@ -120,7 +102,7 @@ def session(self):
         return session
 
     def recycle(self, session):
```

Comment:
Unrelated to this PR - but how come this is a public method, rather than having `Session` return itself to the pool on `close`? Doing it on close is usually helpful, since then you only have to have the session to recycle it, rather than also pass around the graph database instance. 

Also a bit surprised it's on GraphDatabase, rather than on Driver - does that mean Driver instances share access to a process-global pool?